### PR TITLE
Fix duckValueAt for INTERVAL_DAY_TIME

### PR DESCRIPTION
**Pull request by aditi-pandit**
Follow-up from https://github.com/facebookincubator/velox/pull/7806#discussion_r1411131337

### CODE REVIEW

**Review by mbasmanova — COMMENTED** [review]
@aditi-pandit Thank you for the fix. Would it be possible to add a small test to exercise this code path to make sure it works and prevent accidental breakage in the future?

**Comment by aditi-pandit** [comment]
> @aditi-pandit Thank you for the fix. Would it be possible to add a small test to exercise this code path to make sure it works and prevent accidental breakage in the future?

@mbasmanova : Have added a test. PTAL.

File: velox/exec/tests/QueryAssertionsTest.cpp [hunk 1]
```diff
@@ -469,4 +469,28 @@ TEST_F(QueryAssertionsTest, varbinary) {
...
+  // type mapping to BIGINT. Tests its use as a FLAT vector and in a MAP
+  // serialized to DuckDB to cover the specialized code-paths.
+  auto data = makeRowVector(
+      {makeFlatVector<int64_t>({5, 10, 15, 0}, INTERVAL_DAY_TIME())});
+
+  createDuckDbTable({data});
+  auto plan = PlanBuilder().values({data}).planNode();
+  assertQuery(plan, "SELECT * FROM tmp");
+
+  data = makeRowVector({makeNullableMapVector<int64_t, double>(
```
Comment (mbasmanova) [inline velox/exec/tests/QueryAssertionsTest.cpp:483]: nit: perhaps, makeMapVectorFromJson

**Review by mbasmanova — APPROVED** [review]
Thanks.

File: velox/exec/tests/QueryAssertionsTest.cpp (hunk 1 above)
Comment (aditi-pandit) [reply velox/exec/tests/QueryAssertionsTest.cpp]: Done.

**Pull request closed without merging**